Userland/Libraries/LibWeb/SVG/SVGStyleElement.cpp

```diff
@@ -0,0 +1,178 @@
+
+SVGStyleElement::SVGStyleElement(DOM::Document& document, DOM::QualifiedName qualified_name)
+    : SVGElement(document, qualified_name)
+{
+}
+
+SVGStyleElement::~SVGStyleElement() = default;
+
+JS::ThrowCompletionOr<void> SVGStyleElement::initialize(JS::Realm& realm)
+{
+    MUST_OR_THROW_OOM(Base::initialize(realm));
+    set_prototype(&Bindings::ensure_web_prototype<Bindings::SVGStyleElementPrototype>(realm, "SVGStyleElement"));
+
+    return {};
+}
+
+void SVGStyleElement::visit_edges(Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_associated_css_style_sheet.ptr());
+}
+
+void SVGStyleElement::children_changed()
+{
+
+    update_a_style_block();
+    SVGElement::children_changed();
+}
+
+void SVGStyleElement::inserted()
+{
+    update_a_style_block();
+    return SVGElement::inserted();
+}
+
+void SVGStyleElement::removed_from(Node* old_parent)
+{
+    update_a_style_block();
+    return SVGElement::removed_from(old_parent);
+}
+
+// https://www.w3.org/TR/cssom/#remove-a-css-style-sheet
+static void remove_a_css_style_sheet(DOM::Document& document, CSS::CSSStyleSheet& sheet)
+{
+    // 1. Remove the CSS style sheet from the list of document or shadow root CSS style sheets.
+    document.style_sheets().remove_sheet(sheet);
+
+    // 2. Set the CSS style sheet’s parent CSS style sheet, owner node and owner CSS rule to null.
+    sheet.set_parent_css_style_sheet(nullptr);
+    sheet.set_owner_node(nullptr);
+    sheet.set_owner_css_rule(nullptr);
+}
+
+// https://www.w3.org/TR/cssom/#add-a-css-style-sheet
+static void add_a_css_style_sheet(DOM::Document& document, CSS::CSSStyleSheet& sheet)
+{
+    // 1. Add the CSS style sheet to the list of document or shadow root CSS style sheets at the appropriate location. The remainder of these steps deal with the disabled flag.
+    document.style_sheets().add_sheet(sheet);
+
+    // 2. If the disabled flag is set, then return.
+    if (sheet.disabled())
+        return;
+
+    // FIXME: 3. If the title is not the empty string, the alternate flag is unset, and preferred CSS style sheet set name is the empty string change the preferred CSS style sheet set name to the title.
+
+    // FIXME: 4. If any of the following is true, then unset the disabled flag and return:
+    //           The title is the empty string.
+    //           The last CSS style sheet set name is null and the title is a case-sensitive match for the preferred CSS style sheet set name.
+    //           The title is a case-sensitive match for the last CSS style sheet set name.
+
+    // FIXME: 5. Set the disabled flag.
+}
+
+// https://www.w3.org/TR/cssom/#create-a-css-style-sheet
+static void create_a_css_style_sheet(DOM::Document& document, DeprecatedString type, DOM::Element* owner_node, DeprecatedString media, DeprecatedString title, bool alternate, bool origin_clean, DeprecatedString location, CSS::CSSStyleSheet* parent_style_sheet, CSS::CSSRule* owner_rule, CSS::CSSStyleSheet& sheet)
```

Comment:
These are just free functions (not class members), so could be easily shared with the HTMLStyleElement (just make them non-static, and stick the prototypes in some header).